SharpMap/Layers/VectorLayer.cs

```diff
@@ -236,7 +236,7 @@ public override void Render(Graphics g, MapViewport map)
         /// <param name="map">The map object</param>
         /// <param name="envelope">The envelope to render</param>
         /// <param name="theme">The theme to apply</param>
-        protected void RenderInternal(Graphics g, MapViewport map, Envelope envelope, ITheme theme)
+        protected void RenderInternal(Graphics g, MapViewport map, Envelope envelope, IThemeEx theme)

```

Comment:
Don't change the signature of the function, instead:
* Add theme evaluators:
```C#
#region Theme evaluators
private abstract class ThemeEvaluatorBase
{
    public abstract IStyle GetStyle(MapViewport mvp, FeatureDataRow feature);
}

private class ThemeEvaluator : ThemeEvaluatorBase
{
    private readonly ITheme _theme;

    public ThemeEvaluator(ITheme theme)
    {
        _theme = theme;
    }
    public sealed override IStyle GetStyle(MapViewport mvp, FeatureDataRow feature)
    {
        return _theme.GetStyle(feature);
    }
}

private class ThemeExEvaluator : ThemeEvaluatorBase
{
    private readonly IThemeEx _theme;

    public ThemeExEvaluator(IThemeEx theme)
    {
        _theme = theme;
    }
    public sealed override IStyle GetStyle(MapViewport mvp, FeatureDataRow feature)
    {
        return _theme.GetStyle(mvp, feature);
    }
}
#endregion
```
* Add eval style delegate to RenderInternal
```C#
    Func<MapViewport, FeatureDataRow, IStyle> evalStyle;
    if (themeEx != null)
        evalStyle = new ThemeExEvaluator(themeEx).GetStyle;
    else
        evalStyle = new ThemeEvaluator(theme).GetStyle;
```
* use that instead of calling theme.GetStyle directly.
```C#
var outlineStyle = evalStyle(map, feature) as VectorStyle;
```